status-im/status-react
# Add reading nodes from contract

This PR is part of network incentivisation. It adds a way for a client
to pull nodes from a contract.

This is done by selecting the `eth.contract` fleet. If that is selected
on login it will fetch nodes from a contract and pass them to status-go.
If these can't be fetched, it will default to `eth.beta`.

Currently contract information are hard-coded, but eventually the user
will be able to add their own (probably).

Toggled off in release.

There's no discovery, only static nodes for nodes for now.

### Testing

1) Go to fleets
2) Select `eth.contract`
3) Login
4) Everything should work as expected (roughly)

If login is when offline it might throw an exception `unable to find peer with id`, that's something it will be fixed at later stages.

Upgrades status-go, small changes but good to do some sanity checks.


status: ready
